src/bank.rs

```diff
@@ -668,6 +699,21 @@ mod tests {
         assert_eq!(bank.get_balance(&pubkey), 500);
     }
 
+    #[test]
+    fn test_user_data() {
+        let mint = Mint::new(10_000);
+        let bank = Bank::new(&mint);
+        let pubkey = mint.keypair().pubkey();
+
+        let mut tx = Transaction::new(&mint.keypair(), pubkey, 0, bank.last_id());
```

Comment:
There's currently a `Transaction::new_from_instruction()`. Sounds like there should be a `Transaction::new_from_userdata()`?